umich_api/api_utils.py

```diff
@@ -114,21 +134,69 @@ def _api_call(self, api_call, access_token, method="GET", payload=None):
             return resp
 
     def get_next_page(self, response):
+        #type: (requests.Response) -> Dict
         if not 'next' in response.links:
             return None
         self.__log.debug(response.links)
         return parse_qs(urlparse(response.links['next']['url']).query)
 
     def get_access_token(self, token_url, client_scope):
-        # type: (str) -> str
-        """Retrieves an access token from the specified URL
-        
+        # type: (str, str) -> str
+        """Retrieves access token from the local tokens dictionary
+           If it's expired it will attempt to renew it
         :param token_url: Full URL to retrieve access token
         :type token_url: str
         :param client_scope: Client scope to retrieve this token for
         :type client_scope: str
+        :returns Access token str
+        :raises AttributeError If the oauth token could not be found
         """
+        token = self.get_oauth_token(token_url, client_scope)
+        if token is None:
+            raise AttributeError(f"Could not find oauth token for {token_url} {client_scope}")
```

Comment:
The caller method of this is not catching the exception to raise it to its caller function, I don't know if that is a good or bad way of program flow. But definitely, the whoever is calling should put the code in the `try/catch.` The test case 'test_token_renewal' is not doing it and ending up in exception instead of the assertion failure.

Reply:
The only way you'd get this exception is if you passed a scope or url that was invalid. I don't believe that any of the test cases do this. So this would be an application level error.